src/app/OperationalSessionSetup.cpp

```diff
@@ -361,12 +363,11 @@ void OperationalSessionSetup::CleanupCASEClient()
 
 void OperationalSessionSetup::OnSessionReleased()
 {
-    MoveToState(State::HasAddress);
-}
-
-void OperationalSessionSetup::OnFirstMessageDeliveryFailed()
-{
-    LookupPeerAddress();
+    // This will not be called when we are in the process of establishing a session. This will
```

Comment:
I'm kind of curious: what calls this in the address lookup case?

Reply:
Sorry I was in some meetings. I don't know how often it will actually happen, but I am assuming that it is theoretically possible that the session we are trying to do a lookup on will have a MarkForEviction() called on it while we are waiting for the address to resolve, right?

Reply:
Ah, yes, that would do it.